chore: automate-release-process (#1027)
kubernetes-monitoring/kubernetes-mixin: README.md

A set of Grafana dashboards and Prometheus alerts for Kubernetes.

## Releases

> > Note: Releases up until `release-0.12` are changes in their own branches. There are changelogs in releases starting from `version-0.13.0`. 

Reviewer: ```suggestion
> Note: Releases up until `release-0.12` are changes in their own branches. There are changelogs in releases starting from [version-0.13.0](https://github.com/kubernetes-monitoring/kubernetes-mixin/releases/tag/version-0.13.0).
```

Warning: By default the expressions will generate *grafana 7.2+* compatible rules using the *$__rate_interval* variable for rate functions. If you need backward compatible rules please set *grafana72: false* in your *_config*

### Release steps

> Maintainers can run release when it's time to do so. 

Reviewer: ```suggestion
Maintainers can trigger the [release workflow](.github/workflows/release.yaml) by pushing a git tag that matches the pattern: `version-*`.
```

1. Checkout `master` branch and pull for latest.
    ```bash
    git checkout master
    ```
2. Create a tag following sem-ver versioning for the version and trigger release. 
    ```bash

>     tag=version=${x.x.x}; git tag $tag && git push origin $tag

Reviewer: Change to indicate required pattern and make it clear that there are 3 independent components of the version number:

```suggestion
	# replace MAJOR.MINOR.PATCH with e.g. 1.2.3
    tag=version=version-MAJOR.MINOR.PATCH; git tag $tag && git push origin $tag
```

    ```

#### Decisions on backfilling releases

> We wanted to backfill `release-0.1` to `release-0.12` to have a changelog, but we were not able to use a github action in a newer commit to trigger a release that generates a changelog on older commits. We have also tried adding a github action in existing release branches to trigger a release with changelog when a tag is created and pushed in a branch, but the generated changelog was empty as there were no previous releases. Since `release-0.12` was updated 3 years ago, we decided to start release and changelog from `version-0.13.0`

Reviewer: ```suggestion
We wanted to backfill `release-0.1` to `release-0.12` to have a changelog, but we were not able to use a GitHub action in a newer commit to trigger a release that generates a changelog on older commits. See #489 for full discussion.
```
